File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/model/realtimealert/RealTimeStatus.java
package com.skedgo.tripkit.common.model.realtimealert;

import android.text.TextUtils;

import java.util.Locale;

import androidx.annotation.Nullable;

public enum RealTimeStatus {
    CAPABLE, IS_REAL_TIME, INCAPABLE;

    @Nullable
    public static RealTimeStatus from(String s) {
        if (TextUtils.isEmpty(s)) {
            return null;
        }

        final String lowerCase = s.toLowerCase(Locale.US);
        for (RealTimeStatus value : values()) {
            if (TextUtils.equals(lowerCase, value.name().toLowerCase(Locale.US))) {
                return value;
            }
        }

        return null;
    }
}

File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/model/realtimealert/RealTimeStatus.kt
package com.skedgo.tripkit.common.model.realtimealert

import android.text.TextUtils

enum class RealTimeStatus {
    CAPABLE, IS_REAL_TIME, INCAPABLE;

    companion object {
        @JvmStatic
        fun from(s: String?): RealTimeStatus? {
            if (s == null || TextUtils.isEmpty(s)) {
                return null
            }

            val lowerCase = s.lowercase()
            for (value in values()) {
                if (TextUtils.equals(lowerCase, value.name.lowercase())) {
                    return value
                }
            }

            return null
        }
    }
}